### ISSUE TITLE
util-linux bash-completion conflict

### ISSUE THREAD

**Issue by sambrightman** [Linuxbrew/homebrew-core#7624]
**Please note we will close your issue without comment if you delete, do not read or do not fill out the issue checklist below and provide ALL the requested information. If you repeatedly fail to use the issue template, we will block you from ever submitting issues to Homebrew again.**

- [X] are reporting a bug others will be able to reproduce and not asking a question. If you're not sure or want to ask a question do so on our Discourse: https://discourse.brew.sh
- [X] have a problem with `brew install` (or `upgrade`, `reinstall`) a single, official formula (not cask)? If it's a general `brew` problem please file this issue at Homebrew/brew: https://github.com/Homebrew/brew/issues/new/choose. If it's a `brew cask` problem please file this issue at https://github.com/caskroom/homebrew-cask/issues/new. If it's a tap (e.g. Homebrew/homebrew-php) problem please file this issue at the tap.
- [X] ran `brew update` and can still reproduce the problem?
- [X] ran `brew doctor`, fixed all issues and can still reproduce the problem?

Except HOMEBREW_BUILD_FROM_SOURCE and presence of `cargo-install-update-config` in `$PATH`.

- [X] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
- [X] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

https://gist.github.com/f83cfe9b66037f65584ae6d6c4735ae3

To help us debug your issue please explain:
- What you were trying to do (and why)
- What happened (include command output)
- What you expected to happen
- Step-by-step reproduction instructions (by running `brew install` commands)

Trying to upgrade util-linux, it fails due to permisssion denied on the system bash-completions directory. I have bash-completion installed via Linuxbrew with bash completions and this worked in the past:

```
[sbrightman@sbrightmanlinux ~]$ brew info util-linux
util-linux: stable 2.32 (bottled)
Collection of Linux utilities
https://github.com/karelzak/util-linux
/spare/ssd/sbrightman/dev/linuxbrew/Cellar/util-linux/2.30.1 (361 files, 31.7MB) *
  Built from source on 2017-11-01 at 12:17:27
From: https://github.com/Linuxbrew/homebrew-core/blob/master/Formula/util-linux.rb
==> Dependencies
Required: ncurses ✔
Optional: linuxbrew/extra/linux-pam ✘
==> Options
--with-linux-pam
	Build with linux-pam support
==> Caveats
Bash completion has been installed to:
  /spare/ssd/sbrightman/dev/linuxbrew/etc/bash_completion.d
[sbrightman@sbrightmanlinux ~]$ 
```

If I run `--debug` and start a shell, `/usr/bin/pkg-config` is being called and doesn't figure out the Linuxbrew location for completions. I have `pkg-config` installed via Linuxbrew and running outside of the shell returns the correct completions location. Adding `depends_on bash-completion` doesn't help. 

I expect there is a way to disable the completions via configure but a) I want them; b) it seems like this shouldn't be happening. The conflict was previously mentioned in Linuxbrew/legacy-linuxbrew#1114.

I cannot install Linuxbrew in the recommended location and I cannot remove the system bash-completions.

**Comment by iMichka** [comment]
@sambrightman Can you try out this patch: #7672 ?

**Comment by sambrightman** [comment]
Yes, that works.

**Comment by iMichka** [comment]
Fix by #7672
